### PR TITLE
Feature/validator

### PR DESCRIPTION
**Pull request by ramalaso**
Please review.

### CODE REVIEW

File: src/main/java/com/jalasoft/webservice/controller/AudioController.java [hunk 1]
```diff
@@ -31,6 +35,8 @@
 @RestController
 @RequestMapping("/api/v1.0/audioConv")
 public class AudioController {
+    Logger logger = LoggerFactory.getLogger(AudioController.class);
+    Response response = new Response();
     /**
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/controller/AudioController.java:40]: add space between the comments and the Respondes

File: src/main/java/com/jalasoft/webservice/controller/AudioController.java [hunk 2]
```diff
@@ -52,18 +58,25 @@ public Response convert (@RequestParam("file") MultipartFile file,
                              @RequestParam(value = "bitRate", defaultValue = "64000") int bitRate,
                              @RequestParam(value = "channels", defaultValue = "1") int channels,
                              @RequestParam(value = "samplingRate", defaultValue = "22050") int samplingRate,
-                             @RequestParam(value = "format", defaultValue = "mp3") String format)
-                             throws IOException, TikaException, SAXException, NoSuchAlgorithmException {
+                             @RequestParam(value = "format", defaultValue = "mp3") String format) throws IOException, TikaException, SAXException, NoSuchAlgorithmException {
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/controller/AudioController.java]: this line oveload the 100 characters

File: src/main/java/com/jalasoft/webservice/controller/AudioController.java [hunk 3]
```diff
@@ -52,18 +58,25 @@ public Response convert (@RequestParam("file") MultipartFile file,
...
+        logger.info("Starting Audio Controller - Method: " + new Object() {}.getClass().getEnclosingMethod().getName());
+        Validator validator = new Validator();
+        if (!validator.isValidAudio(file.getOriginalFilename())){
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/controller/AudioController.java]: insert space between (){

File: src/main/java/com/jalasoft/webservice/controller/AudioController.java [hunk 4]
```diff
@@ -82,10 +95,11 @@ public Response convert (@RequestParam("file") MultipartFile file,
             cri.setSamplingRate(samplingRate);
             cri.setFormat(format);
         } else {
+            logger.error("The cheksum send is not match - Method: " + new Object() {}.getClass().getEnclosingMethod().getName());
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/controller/AudioController.java]: review if this line do not super the 100 characters

File: src/main/java/com/jalasoft/webservice/controller/DownloadController.java [hunk 5]
```diff
@@ -27,13 +29,14 @@
 @RestController
 @RequestMapping("/download")
 public class DownloadController {
-
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/controller/DownloadController.java:30]: insert space between the first line and the method

File: src/main/java/com/jalasoft/webservice/controller/ImageController.java [hunk 6]
```diff
@@ -70,6 +74,7 @@ public Response Convert (@RequestParam("file") MultipartFile file,
                 imageCriteria.setDestinationPath(utils.getPublic());
                 imageCriteria.setExtension(ext);
             } else {
+                logger.error("The cheksum send is not match - Method: " + new Object() {}.getClass().getEnclosingMethod().getName());
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/controller/ImageController.java]: review this line do not overload the 100 character

File: src/main/java/com/jalasoft/webservice/controller/ImageController.java [hunk 7]
```diff
@@ -46,6 +48,7 @@ public Response Convert (@RequestParam("file") MultipartFile file,
                              @RequestParam (value = "dpi", defaultValue = "") int dpi,
                              @RequestParam(value = "ext", defaultValue = "") String ext) throws IOException,
                              NoSuchAlgorithmException, TikaException, SAXException {
+            logger.info("Starting Image Controller - Method: " + new Object() {}.getClass().getEnclosingMethod().getName());
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/controller/ImageController.java]: review this line do not overload the 100 character

File: src/main/java/com/jalasoft/webservice/controller/OCRController.java [hunk 8]
```diff
@@ -42,14 +45,16 @@ public Response OCRExtractor (@RequestParam("file") MultipartFile file,
...
         String filePath = FileManager.getFilePath(file);
         Checksum checksum1 = new Checksum();
         Response test = new Response();
         DBManager db = new DBManager();
         String checksumResult = checksum1.checksum(filePath);
         String pathDb = "";
-        if (checksum.equals(checksumResult)) {
-            if (db.getPath(checksumResult).isEmpty()) {
+        if (checksum.equals(checksumResult)){
+            logger.error("The cheksum send is not match - Method: " + new Object() {}.getClass().getEnclosingMethod().getName());
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/controller/OCRController.java]: review this line do not overload the 100 character

File: src/main/java/com/jalasoft/webservice/controller/OCRController.java [hunk 9]
```diff
@@ -42,14 +45,16 @@ public Response OCRExtractor (@RequestParam("file") MultipartFile file,
...
+            if (db.getPath(checksumResult).isEmpty()){
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/controller/OCRController.java]: insert space between (){

File: src/main/java/com/jalasoft/webservice/model/AudioConvert.java [hunk 10]
```diff
@@ -55,9 +61,11 @@ public Response convert(Criteria criteria) throws IOException {
             String [] filePaths = new String[5];
             filePaths[0]=audiocri.getTarget();
             zipFiles.zipFiles(filePaths);
+            logger.info("Audio conversion succesfully - Method: " + new Object() {}.getClass().getEnclosingMethod().getName());
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/model/AudioConvert.java]: review this line do not overload the 100 character

**Review by FerHinojosa — CHANGES_REQUESTED** [review]
Please review the comments

File: src/main/java/com/jalasoft/webservice/controller/AudioController.java (hunk 1 above)
Comment (ramalaso) [reply src/main/java/com/jalasoft/webservice/controller/AudioController.java:40]: Done

File: src/main/java/com/jalasoft/webservice/controller/AudioController.java (hunk 2 above)
Comment (ramalaso) [reply src/main/java/com/jalasoft/webservice/controller/AudioController.java]: Done

File: src/main/java/com/jalasoft/webservice/controller/AudioController.java (hunk 3 above)
Comment (ramalaso) [reply src/main/java/com/jalasoft/webservice/controller/AudioController.java]: Done

File: src/main/java/com/jalasoft/webservice/controller/AudioController.java (hunk 4 above)
Comment (ramalaso) [reply src/main/java/com/jalasoft/webservice/controller/AudioController.java]: Done

File: src/main/java/com/jalasoft/webservice/controller/DownloadController.java (hunk 5 above)
Comment (ramalaso) [reply src/main/java/com/jalasoft/webservice/controller/DownloadController.java:30]: Done

File: src/main/java/com/jalasoft/webservice/controller/ImageController.java (hunk 7 above)
Comment (ramalaso) [reply src/main/java/com/jalasoft/webservice/controller/ImageController.java]: Done

File: src/main/java/com/jalasoft/webservice/controller/ImageController.java (hunk 6 above)
Comment (ramalaso) [reply src/main/java/com/jalasoft/webservice/controller/ImageController.java]: Done

File: src/main/java/com/jalasoft/webservice/controller/OCRController.java (hunk 8 above)
Comment (ramalaso) [reply src/main/java/com/jalasoft/webservice/controller/OCRController.java]: Done

File: src/main/java/com/jalasoft/webservice/controller/OCRController.java (hunk 9 above)
Comment (ramalaso) [reply src/main/java/com/jalasoft/webservice/controller/OCRController.java]: Done

File: src/main/java/com/jalasoft/webservice/model/AudioConvert.java (hunk 10 above)
Comment (ramalaso) [reply src/main/java/com/jalasoft/webservice/model/AudioConvert.java]: Done

**Review by ramalaso — COMMENTED** [review]
Changes done.

File: src/main/java/com/jalasoft/webservice/model/OCRCriteria.java [hunk 11]
```diff
@@ -48,4 +49,23 @@ public void setLang(String lang) {
         this.lang = lang;
         return;
     }
+
+    @Override
+    public void Validate() throws ParamInvalidException {
+        if(this.filePath==null){
+            throw new ParamInvalidException(10, "filePath");
+        }
+
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/model/OCRCriteria.java]: remove this space

File: src/main/java/com/jalasoft/webservice/model/OCRCriteria.java [hunk 12]
```diff
@@ -48,4 +49,23 @@ public void setLang(String lang) {
...
+        if (this.filePath.isEmpty()){
+            throw new ParamInvalidException(11, "filePath");
+        }
+
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/model/OCRCriteria.java]: remove this space

File: src/main/java/com/jalasoft/webservice/model/OCRCriteria.java [hunk 13]
```diff
@@ -48,4 +49,23 @@ public void setLang(String lang) {
...
+        if(this.lang == null){
+            throw new ParamInvalidException(10, "lang");
+        }
+
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/model/OCRCriteria.java]: remove this space

File: src/main/java/com/jalasoft/webservice/model/OCRCriteria.java [hunk 14]
```diff
@@ -48,4 +49,23 @@ public void setLang(String lang) {
         this.lang = lang;
         return;
     }
+
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/model/OCRCriteria.java:52]: This method does not have documentation

File: src/main/java/com/jalasoft/webservice/model/OCRCriteria.java [hunk 15]
```diff
@@ -16,6 +16,7 @@
  * @version v1.0
  */
 public class OCRCriteria extends Criteria {
+
     private String lang;
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/model/OCRCriteria.java:21]: remove this space

File: src/main/java/com/jalasoft/webservice/model/Criteria.java [hunk 16]
```diff
@@ -34,4 +33,6 @@ public String getFilePath() {
     public void setFilePath(String filePath) {
         this.filePath = filePath;
     }
+
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/model/Criteria.java:36]: Please documented this method

File: src/main/java/com/jalasoft/webservice/controller/AudioController.java [hunk 17]
```diff
@@ -61,10 +61,10 @@ public Response convert (@RequestParam("file") MultipartFile file,
                              @RequestParam(value = "format", defaultValue = "mp3") String format) throws IOException, TikaException, SAXException, NoSuchAlgorithmException {
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/controller/AudioController.java]: Review this line do not overload the 100 words.

**Review by FerHinojosa — CHANGES_REQUESTED** [review]
Please review the commends

File: src/main/java/com/jalasoft/webservice/controller/OCRController.java [hunk 18]
```diff
@@ -44,7 +41,7 @@
     public Response OCRExtractor (@RequestParam("file") MultipartFile file,
                                   @RequestParam(value = "checksum",defaultValue = "false")String checksum,
                                   @RequestParam(value = "lang", defaultValue = "") String lang) throws IOException,
-                                  NoSuchAlgorithmException {
+            NoSuchAlgorithmException, ParamInvalidException {
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/controller/OCRController.java]: Correct this line

File: src/main/java/com/jalasoft/webservice/model/AudioCriteria.java [hunk 19]
```diff
@@ -118,4 +120,52 @@ public String getFormat() {
...
+     * @author Raul Laredo
+     * @version v1.0
+     */
+    @Override
+    public void Validate() throws ParamInvalidException {
+        String ext =  FilenameUtils.getExtension(this.filePath);
+        if(this.filePath==null){
+            throw new ParamInvalidException(10, "filePath");
+        }
+
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/model/AudioCriteria.java]: remove the space please

**Review by FerHinojosa — CHANGES_REQUESTED** [review]

File: src/main/java/com/jalasoft/webservice/controller/ImageController.java [hunk 20]
```diff
@@ -59,7 +59,8 @@ public Response Convert (@RequestParam("file") MultipartFile file,
             ImageCriteria imageCriteria = new ImageCriteria();
             Utils utils = new Utils();
             if (metadata) {
-                logger.info("Verifying metadata - Method: " + new Object() {}.getClass().getEnclosingMethod().getName());
+                logger.info("Verifying metadata - Method: " +
+                new Object() {}.getClass().getEnclosingMethod().getName());
                 MetadataFileCreator metadataF =  new MetadataFileCreator();
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/controller/ImageController.java]: In this line between equals and new we have two spaces, please remove one

File: src/main/java/com/jalasoft/webservice/controller/ImageToImageController.java [hunk 21]
```diff
@@ -62,7 +62,8 @@ public Response Convert (@RequestParam("file") MultipartFile file,
         ImageToImageCriteria imageToImageCriteria = new ImageToImageCriteria();
         Utils utils = new Utils();
         if (metadata) {
-            logger.info("Verifying metadata - Method: " + new Object() {}.getClass().getEnclosingMethod().getName());
+            logger.info("Verifying metadata - Method: " +
+            new Object() {}.getClass().getEnclosingMethod().getName());
             MetadataFileCreator metadataF =  new MetadataFileCreator();
```
Comment (FerHinojosa) [inline src/main/java/com/jalasoft/webservice/controller/ImageToImageController.java:67]: In this line between equals and new we have two spaces, please remove one

**Review by FerHinojosa — CHANGES_REQUESTED** [review]

**Review by FerHinojosa — APPROVED** [review]

**Pull request merged**